pkg/reconciler/v1alpha1/revision/resources/deploy_test.go

```diff
@@ -1164,53 +730,12 @@ func TestMakeDeployment(t *testing.T) {
 				TimeoutSeconds: 45,
 			},
 		},
-		lc: &logging.Config{},
-		nc: &config.Network{},
-		oc: &config.Observability{},
-		ac: &autoscaler.Config{},
-		cc: &config.Controller{},
-		want: &appsv1.Deployment{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: "foo",
-				Name:      "bar-deployment",
-				Labels: map[string]string{
-					serving.RevisionLabelKey: "bar",
-					serving.RevisionUID:      "1234",
-					AppLabelKey:              "bar",
-				},
-				Annotations: map[string]string{},
-				OwnerReferences: []metav1.OwnerReference{{
-					APIVersion:         v1alpha1.SchemeGroupVersion.String(),
-					Kind:               "Revision",
-					Name:               "bar",
-					UID:                "1234",
-					Controller:         &boolTrue,
-					BlockOwnerDeletion: &boolTrue,
-				}},
-			},
-			Spec: appsv1.DeploymentSpec{
-				Replicas: &one,
-				Selector: &metav1.LabelSelector{
-					MatchLabels: map[string]string{
-						serving.RevisionUID: "1234",
-					},
-				},
-				ProgressDeadlineSeconds: &ProgressDeadlineSeconds,
-				Template: corev1.PodTemplateSpec{
-					ObjectMeta: metav1.ObjectMeta{
-						Labels: map[string]string{
-							serving.RevisionLabelKey: "bar",
-							serving.RevisionUID:      "1234",
-							AppLabelKey:              "bar",
-						},
-						Annotations: map[string]string{
-							sidecarIstioInjectAnnotation: "true",
-						},
-					},
-					// Spec: filled in below by makePodSpec
-				},
-			},
-		},
+		lc:   &logging.Config{},
+		nc:   &config.Network{},
+		oc:   &config.Observability{},
+		ac:   &autoscaler.Config{},
+		cc:   &config.Controller{},
```

Comment:
We should change the test kernel to default these when nil (or maybe just make them non-pointers in the row struct and let Go do it for us?)

Reply:
Sure, I'll tackle that in a follow up PR.